src/main/java/org/kainos/ea/api/EmployeeService.java

```diff
@@ -9,9 +13,24 @@
 public class EmployeeService {
 
     private EmployeeDao employeeDao = new EmployeeDao();
+    private EmployeeRequestValidator employeeRequestValidator = new EmployeeRequestValidator();
 
     public List<Employee> getAllEmployee() throws SQLException {
 
         return employeeDao.getAllEmployees();
     }
+
+    public int createNewEmployee(EmployeeRequest employeeRequest) throws EmployeeRequestIsNotValid, SQLException {
+
+        try {
+            String isEmployeeRequestValid = employeeRequestValidator.isEmployeeValid(employeeRequest);
+
+            if (isEmployeeRequestValid != null) {
+                throw new EmployeeRequestIsNotValid(isEmployeeRequestValid);
+            }
+            return employeeDao.createNewEmployee(employeeRequest);
+        } catch (DeliveryEmployeeCouldNotBeCreatedException e) {
+            throw new EmployeeRequestIsNotValid(e.getMessage());
+        }
```

Comment:
the DeliveryEmployeeCouldNotBeCreatedException is being thrown by the dao if connection to db fails so should return a 500 from controller however you're throwing a EmployeeRequestIsNotValid which is returning a 400 from controller

Reply:
Should be fixed now! If any SQL exceptions happen, we create a new sqlexception. Which should be now return a 500.